lib/payload.json

```diff
@@ -137,6 +166,44 @@
             }
           },
           "additionalProperties": false
+        },
+        "water": {
+          "type": "object",
+          "properties": {
+            "leak": {
+              "description": "Water leaking (string)",
+              "$ref": "#/definitions/leak"
+            }
+          },
+          "additionalProperties": false
+        },
+        "action": {
+          "type": "object",
+          "properties": {
+            "motionState": {
+              "description": "State of Motion (number)",
+              "$ref": "#/definitions/state"
+            },
+            "doorState": {
+              "description": "State of a door",
+              "$ref": "#/definitions/doorState"
+            }
+          },
+          "additionalProperties": false
+        },
+        "position": {
+          "type": "object",
+          "properties": {
+            "latitude": {
+              "description": "Horizontal distance from equator",
+              "$ref": "#/definitions/latitude"
+            },
+            "longitude": {
+              "description": "Vertical distance from prime meridian",
+              "$ref": "#/definitions/longitude"
+            }
```

Comment:
I don't think they'll be reused so you can just declare them here.

The reason why some other quantities (like percentage) are defined top-level is because they're easily reused in other places.

Reply:
Alright i guess i got confused there. I followed the schema as it was so i assumed every object needed to be defined top-level. Will change it 👍